### PR TITLE
Make metax errors visible

### LINKED ISSUE #382: Metax error handling backend/frontend
There needs to be a way to identify where from error is coming, is it Metax or an internal error.
One idea would be to add to errors coming from Metax service a suffix indicating that this is an error from an external service. Also maybe some informative text on how to deal with this error.

### PR DESCRIPTION
**Pull request by csc-felipe**
### Description

Refactored Metax error handling, so error becomes visible to end user, only in publication endpoint.

### Related issues

- Fixes #382

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Changes Made

- Remove lingering error page references
- Remove broken HEAD request to check for connection - Metax doesn't support HEAD requests
- Remove duplicate code
- Let Metax errors surface
- Catches unexpected errors and return JSON Problem detail
- Update Docker compose version to 3.4 as the `build target` parameter was introduced in that version
- Add Metax connection check to more endpoints.
- Ignore and log metax errors in Object API endpoints

### Testing

Existing tests should continue to work.

- [x] Tests do not apply

### CODE REVIEW

**Comment by csc-felipe** [comment]
I wonder why integration tests fail. They all pass in dev env.

**Comment by genie9** [comment]
could you please change the base to branch `refactor/json-error-handling` for now until we sort out what to do with braking changes caused to frontend

**Comment by csc-felipe** [comment]
> could you please change the base to branch `refactor/json-error-handling` for now until we sort out what to do with braking changes caused to frontend

Done

**Review by genie9 — CHANGES_REQUESTED** [review]

File: metadata_backend/api/handlers/submission.py [hunk 1]
```diff
@@ -564,8 +564,6 @@ async def publish_submission(self, req: Request) -> Response:
 
         await self._handle_check_ownership(req, "submissions", submission_id)
 
-        await metax_handler.check_connection()
```
[the remark this answers is not in the document]
Comment (csc-felipe) [reply metadata_backend/api/handlers/submission.py:503]: This doesn't guarantee that the next request will succeed. But sure, we can leave it there, and make a better solution in another change.

**Review by genie9 — CHANGES_REQUESTED** [review]

**Review by genie9 — APPROVED** [review]

**Pull request merged**